tsconfig.json

```diff
@@ -1,16 +1,13 @@
 {
   "compilerOptions": {
-    "module": "commonjs",
-    "lib": ["es6"],
+    "module": "node16",
+    "moduleResolution": "node16",
```

Comment:
Sorry, one more detail. Since the lowest supported Node.js version is 20, having module resolution set like this will handle self-referencing. Which is importing `sort-package-json` directly without `paths` (deleted below).

Reply:
> using ESM syntax

We also bundle the CommonJS version, should we test it? https://github.com/keithamus/sort-package-json/blob/aa6774ad937feb83178c8bc981f08305e1d22b5c/package.json#L27

Reply:
`publint` seems unhappy with the types. https://publint.dev/sort-package-json@3.2.1

Reply:
Ah.. This is out of scope for TSTyche. It only checks correctness types.

`publint` or `@arethetypeswrong/cli` can help with checking `package.json`. They both are unhappy. Here is result from `attw`: https://arethetypeswrong.github.io/?p=sort-package-json%403.2.1

By the way, `publint` recommends using both of these tools together ([source](https://publint.dev/docs/comparisons#are-the-types-wrong)). That sounds too much. Or? What you think?

I can also revert the `tsconfig.json` and this PR can land. The rest can be solved separately.